client/src/components/NavigationBar.tsx

```diff
@@ -2,13 +2,33 @@ import { Container, Nav, Navbar } from "react-bootstrap";
 import { useLocation, useNavigate } from "react-router-dom";
 import { PersonCircle } from "react-bootstrap-icons";
 import { NAV_CONFIG } from "../App.tsx";
+import { useEffect, useState } from "react";
 
 function NavigationBar() {
   const location = useLocation().pathname;
+  const [username, setUsername] = useState<string | null>(null);
   const navigate = useNavigate();
 
+  //fetch the username from the server
+  useEffect(() => {
+    fetch("/api/users/whoami")
```

Comment:
I think `whoami` is better, because the route returns just a username string, not a user object (as you would expect from `users/current`). It returns a string, as we only need the username to setup the profile page link. The profile page then fetches the actual user object. 

`users/me` doesn't sound well in my opinion.